plugins/gcalendar/src/components/CalendarCard/CalendarCard.tsx

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { sortBy } from 'lodash';
+import { DateTime } from 'luxon';
+import React, { useEffect, useMemo, useState } from 'react';
+
+import { InfoCard, Progress } from '@backstage/core-components';
+import { useAnalytics } from '@backstage/core-plugin-api';
+
+import { Box, IconButton, Typography } from '@material-ui/core';
+import PrevIcon from '@material-ui/icons/NavigateBefore';
+import NextIcon from '@material-ui/icons/NavigateNext';
+
+import {
+  useCalendarsQuery,
+  useEventsQuery,
+  useSignIn,
+  useStoredCalendars,
+} from '../../hooks';
+import calendarCardIcon from '../../icons/calendarIcon.svg';
+import { CalendarEvent } from './CalendarEvent';
+import { CalendarSelect } from './CalendarSelect';
+import { SignInContent } from './SignInContent';
+import { getStartDate } from './util';
+
+export const CalendarCard = () => {
```

Comment:
Looking at this is is a Card based component, I'm just curious in knowing if found any limitations in the `createCardExtension` extension creator that we provide as a utility for creating home page components, see more here https://github.com/backstage/backstage/blob/master/plugins/home/README.md#creating-components.

Reply:
It's just because of custom card header :) `createCardExtension ` only allows title string for the header, but in this widget the header is used to display and change selected date and to select calendars. And the state also should be shared between header and content, but `ContextProvider` is provided only for content  inside `createCardExtension`.

Reply:
Got it, that is great feedback, thank you! We might look into how we can improve that to cover a bit more flexibility on the cards (but still keep consistent)